source/taskmanager/src/main/java/nl/aerius/taskmanager/mq/RabbitMQChannelQueueEventsWatcher.java

```diff
@@ -0,0 +1,131 @@
+ */
+package nl.aerius.taskmanager.mq;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabbitmq.client.AMQP;
+import com.rabbitmq.client.Channel;
+import com.rabbitmq.client.Connection;
+import com.rabbitmq.client.Consumer;
+import com.rabbitmq.client.DefaultConsumer;
+import com.rabbitmq.client.Envelope;
+import com.rabbitmq.client.ShutdownSignalException;
+
+import nl.aerius.taskmanager.adaptor.WorkerSizeObserver;
+import nl.aerius.taskmanager.adaptor.WorkerSizeProviderProxy;
+import nl.aerius.taskmanager.client.BrokerConnectionFactory;
+
+/**
+ * Watches the RabbitMQ event channel for changes in consumers to dynamically monitor the number of workers added or removed.
+ */
+class RabbitMQChannelQueueEventsWatcher {
+
+  private static final String AMQ_RABBITMQ_EVENT = "amq.rabbitmq.event";
+  private static final String CHANNEL_PATTERN = "consumer.*";
+  private static final String HEADER_PARAM_QUEUE = "queue";
+  private static final String CONSUMER_CREATED = "consumer.created";
+
+  private static final Logger LOG = LoggerFactory.getLogger(RabbitMQChannelQueueEventsWatcher.class);
+
+  private final BrokerConnectionFactory factory;
+  private final WorkerSizeProviderProxy proxy;
+  private Consumer consumer;
+  private Channel channel;
+
+  /**
+   * Constructor.
+   *
+   * @param factory connection factory
+   * @param proxy proxy to get observers for specific worker queues
+   */
+  public RabbitMQChannelQueueEventsWatcher(final BrokerConnectionFactory factory, final WorkerSizeProviderProxy proxy) {
+    this.factory = factory;
+    this.proxy = proxy;
+  }
+
+  /**
+   * Start the watcher.
+   *
+   * @throws IOException
+   */
+  public void start() throws IOException {
+    final Connection c = factory.getConnection();
+    channel = c.createChannel();
+    final String q = channel.queueDeclare().getQueue();
+    channel.queueBind(q, AMQ_RABBITMQ_EVENT, CHANNEL_PATTERN);
+
+    consumer = new DefaultConsumer(channel) {
+      @Override
+      public void handleDelivery(final String consumerTag, final Envelope envelope, final AMQP.BasicProperties properties, final byte[] body)
+          throws IOException {
+        final String event = envelope.getRoutingKey();
+        final Map<String, Object> headers = properties.getHeaders();
+        final Object queue = headers.get(HEADER_PARAM_QUEUE);
+        final String queueName = queue == null ? null : queue.toString();
+        final WorkerSizeObserver observer = proxy.getWorkerSizeObserver(queueName);
+
+        if (observer == null) {
+          LOG.debug("No handler to watch channel changes for queue: " + queueName);
+          return;
+        }
+
+        LOG.trace("Event: {} - queue: {}", event , queueName);
+        if (CONSUMER_CREATED.equals(event)) {
+          observer.onDeltaNumberOfWorkersUpdate(1);
+        }
+        else /* consumer.deleted is the only other possibility */ {
```

Comment:
Bit unusual (for us) to have the else on the new line.

Reply:
That is indeed incorrect. I've also forgot to run spotless...